### PR TITLE
JPERF-457 Install Jira and download dataset at the same time

### PR DESCRIPTION
**Pull request by wyrzyk**

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt [hunk 1]
```diff
@@ -37,53 +40,79 @@ internal class StandaloneNodeFormula(
...
-                jiraIp = ssh.host.ipAddress
+        }
+        val executor = Executors.newCachedThreadPool(
+            ThreadFactoryBuilder()
+                .setNameFormat("install-product-with-dataset-${config.name}-%d")
+                .build()
+        )
+
+        try {
+            val unpackedProductFuture = executor.submit(
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]:
```suggestion
            val unpackedProduct = executor.submitWithLogContext("install product") {
                ssh.newConnection().use { connection ->
                    productDistribution.install(connection, ".")
                }
            }
```

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt [hunk 2]
```diff
@@ -37,53 +40,79 @@ internal class StandaloneNodeFormula(
...
+                    }
+                }
             )
-            connection.execute("echo jira.home=`realpath $jiraHome` > $unpackedProduct/atlassian-jira/WEB-INF/classes/jira-application.properties")
-            connection.execute("echo jira.autoexport=false > $jiraHome/jira-config.properties")
-            downloadMysqlConnector(
-                "https://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-java-5.1.40.tar.gz",
-                connection
+
+            val jiraHomeFuture = executor.submit(
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]:
```suggestion
            val jiraHome = executor.submitWithLogContext("download Jira home") {
                ssh.newConnection().use { connection ->
                    jiraHomeSource.download(connection)
                }
            }
```

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt [hunk 3]
```diff
@@ -37,53 +40,79 @@ internal class StandaloneNodeFormula(
...
+                        }
+                    }
+                }
             )
-            connection.execute("tar -xzf mysql-connector-java-5.1.40.tar.gz")
-            connection.execute("cp mysql-connector-java-5.1.40/mysql-connector-java-5.1.40-bin.jar $unpackedProduct/lib")
-            AwsCli().download(pluginsTransport.location, connection, target = "$jiraHome/plugins/installed-plugins")
+            val jiraHome = jiraHomeFuture.get()
+            val unpackedProduct = unpackedProductFuture.get()
+            ssh.newConnection().use { connection ->
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]: We can extract the lambda contents to a private method to reduce nesting and method height.

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt (hunk 2 above)
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]: This includes `TaskTimer.time`

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt (hunk 3 above)
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]: @wyrzyk WDYT?

Comment (wyrzyk) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]: > @wyrzyk WDYT?

I could refactor the method. OTOH we're going to replace it with a new API soon, so I don't see much value here.

**Comment by wyrzyk** [comment]
```We currently do not have sufficient c5d.9xlarge capacity in the Availability Zone you requested (eu-west-1b). Our system will be working on provisioning additional capacity. You can currently get c5d.9xlarge capacity by not specifying an Availability Zone in your request or choosing eu-west-1a, eu-west-1c. (Service: AmazonEC2; Status Code: 500; Error Code: InsufficientInstanceCapacity; Request ID: be5d959d-df4c-400c-ad6d-64561637acf3)```

File: src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt (hunk 3 above)
Comment (dagguh) [reply src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/StandaloneNodeFormula.kt]: It's less readable than before the PR, so I raised it.
OTOH, it's a small thing, so I will unblock you.

**Comment by dagguh** [comment]
I'm getting this capacity error every single time for at least 6 builds triggered over 3 days now.
It's a constant failure. Perhaps we gotta change the region.

**Comment by wyrzyk** [comment]
> I'm getting this capacity error every single time for at least 6 builds triggered over 3 days now.
> It's a constant failure. Perhaps we gotta change the region.

Or filter out `eu-west-1b` AZ like we do for our internal tests.

**Comment by dagguh** [comment]
Worth a show

File: gradle/dependency-locks/apiDependenciesMetadata.lockfile [hunk 4]
```diff
@@ -14,13 +14,13 @@ com.amazonaws:aws-java-sdk-support:1.11.424
 com.amazonaws:jmespath-java:1.11.424
 com.atlassian.performance.tools:aws-resources:1.4.0
 com.atlassian.performance.tools:concurrency:1.0.0
-com.atlassian.performance.tools:infrastructure:4.10.0
+com.atlassian.performance.tools:infrastructure:4.12.1
```
Comment (wyrzyk) [inline gradle/dependency-locks/apiDependenciesMetadata.lockfile]: Didn't help :(

Comment (dagguh) [reply gradle/dependency-locks/apiDependenciesMetadata.lockfile]: Due to https://ecosystem.atlassian.net/browse/JPERF-461 we cannot see from CI, was it a timeout or not?

**Pull request merged**